### ISSUE TITLE
Dynamic Online Agent Assignment for Ongoing Campaigns

### ISSUE THREAD

**Issue by MichelN89** [chatwoot/chatwoot#8045]
### Is your feature or enhancement related to a problem? Please describe.

Yes, this feature request is related to a specific issue.

Problem Description:
The specific issue at hand is that when creating ongoing campaigns in Chatwoot to send popup messages to website visitors, there is currently no option to select an "online agent" for message delivery. This limitation means that messages can be assigned to agents who may not be available when the message pops up. As a result, there is a risk of delayed or unanswered visitor inquiries due to the lack of real-time agent availability management within ongoing campaigns. This feature enhancement aims to address this issue and ensure a more seamless visitor experience by allowing the selection of an "online agent" for timely responses.

### Describe the solution you'd like

The solution I'd like is for Chatwoot's ongoing campaigns to dynamically select an "online agent" at the moment the campaign is triggered, following a round-robin approach. Additionally, if no agents are online at the time of the campaign activation, the system should fall back on the bot option.

Here's a concise description of the desired solution:

1. When configuring an ongoing campaign in Chatwoot, there should be an option to specify that an "online agent" should be selected dynamically at the time of message delivery.

2. When the campaign is triggered, Chatwoot should automatically check the availability of agents in real-time.

3. The system should then assign the campaign message to the next available agent in a round-robin fashion, ensuring fair distribution.

4. If no agents are online at the moment of campaign activation, there should be a fallback option to use bot.

This comprehensive solution would optimize ongoing campaigns by directing messages to online agents whenever possible, while also offering a seamless fallback to a bot in case of agent unavailability.

### Describe alternatives you've considered

I've also explored the possibility of setting up an automation to assign conversations to an "online agent." However, I found that this option isn't available within the current capabilities of Chatwoot. The absence of this feature makes it challenging to ensure that conversations are consistently assigned to available agents in real-time. Therefore, it further highlights the need for the proposed feature, which would streamline the process of directing messages to online agents without the need for complex workarounds or additional automations.

### Additional context

I've come across a feature in LiveChat that I believe could greatly enhance the ongoing campaign functionality in Chatwoot. You can find information about this feature in the following link: [LiveChat Targeted Messages](https://www.livechat.com/help/targeted-messages/).

**Comment by pranavrajs** [comment]
@MichelN89, could you please provide more details about the problem? The assignment rules are defined in the inbox and should be applied consistently, whether it's a campaign or an inbound conversation. Currently, regardless of where a new conversation is created, it is automatically assigned to an agent who is online in a round robin manner. 

If the assignment is not working, please check if you have the auto-assignment option enabled in the inbox settings.

**Comment by MichelN89** [comment]
@pranavrajs You're right. I probably did something wrong with testing. Sorry for the inconvenience.

**Comment by pranavrajs** [comment]
Closing as there is no activity for a long time.
